Sources_v3/StreamChat/Controllers/EntityDatabaseObserver.swift

```diff
@@ -179,7 +179,11 @@ class EntityDatabaseObserver<Item, DTO: NSManagedObject> {
             forName: DatabaseContainer.WillRemoveAllDataNotification,
             object: context,
             queue: .main
-        ) { [unowned self] _ in
+        ) { [weak self] _ in
+            // Technically, this should rather be `unowned`, however, `deinit` is not always called on the main thread which can
```

Comment:
Why were you expecting `deinit` to be called on main thread?

Reply:
Inexperience? I was young, naive, full of expectations and assumptions of how Swift works.